Feature presidecms 3151 saved filters should namespace their parameters to avoid collisions

jamesallenuk opened the pull request.

jamesallenuk committed:
[twgit] Init feature 'feature-PRESIDECMS-3151_Saved-filters-should-namespace-their-parameters-to-avoid-collisions'.

jamesallenuk committed:
PRESIDECMS-3151 Fix filter collision in PresideObjectService.selectData()

Resolves issue where direct filters and saved filters targeting the same
field would collide, causing parameter overwrite instead of logical AND
behavior. Implements conditional parameter namespacing with 'sf_' prefix
for saved filters when collision detected.

- Only affects actual collision scenarios (direct + saved filters on same field)
- Preserves all existing behavior for non-collision cases
- Handles both struct and string filter formats
- Maintains backward compatibility

sebduggan commented on system/services/presideObjects/PresideObjectService.cfc:
```diff
@@ -3919,9 +3922,71 @@ component displayName="Preside Object Service" {
+				var sqlParts = [];
+				for ( var key in extraFilter.filter ) {
+					if ( StructKeyExists( originalFilterParams, key ) ) {
+						var paramName = "sf_" & ReReplace( key, "[\.\$]", "__", "all" );
```
Any namespacing needs to be unique for each extraFilter, as just "sf_" prefix could still cause collisions between multiple extraFilters that reference the same property (and "sf_" is inaccurate, as it's not just saved filters). Maybe a hash of key and values? And set the value once in each extraFilter iteration.

sebduggan commented on system/services/presideObjects/PresideObjectService.cfc:
```diff
@@ -3919,9 +3922,71 @@ component displayName="Preside Object Service" {
+				}
+			}
+
+			if ( hasCollision ) {
```
Should this all exist in the above "if isStruct()" block?

jamesallenuk committed:
PRESIDECMS-3151 Improved code structure

jamesallenuk committed:
PRESIDECMS-3151 Comprehensive extraFilter collision prevention using unique hashes

Extends original fix to prevent collisions between multiple extraFilters
targeting the same field, not just direct filter vs saved filter scenarios.

jamesallenuk committed:
PRESIDECMS-3151 improved unique param logic

jamesallenuk committed:
PRESIDECMS-3151 only generate hash once per filter

jamesallenuk committed:
PRESIDECMS-3151 added tests

jamesallenuk committed:
PRESIDECMS-3151 added tests using explictely defined params

jamesallenuk committed:
PRESIDECMS-3151 added test using custom name and filter param explicitly defined

jamesallenuk committed:
PRESIDECMS-3151 Fix ambiguous column error by qualifying bare property names in colliding struct extraFilters

When generating SQL for struct extraFilters that collide with direct filters, bare property names (like "id") are now automatically qualified with the table alias (like "object_b.id") to prevent ambiguous column errors when joins are present.
